enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/analytics/AnalyticsFilter.java

```diff
@@ -196,6 +196,13 @@ public void handleSuccessRequest(RequestContext requestContext) {
             requestContext.addMetadataToMap(MetadataConstants.CLIENT_IP_KEY, requestContext.getClientIp());
             requestContext.addMetadataToMap(MetadataConstants.USER_AGENT_KEY,
                     AnalyticsUtils.setDefaultIfNull(requestContext.getHeaders().get("user-agent")));
+
+            // Adding UserName and the APIContext
+            requestContext.addMetadataToMap(MetadataConstants.API_USER_NAME_KEY,
```

Comment:
What happens if username is null. Aren't we setting something like default user?

Reply:
This value could be null if the JWT token does not have a sub claim. In disabled security scenario we set it to "anonymous". may be we could use the same term here to handle username null scenario.

Reply:
shall we set as unknown